components/eam/src/physics/cam/zm_conv_intr.F90

```diff
@@ -1,1341 +1,1193 @@
+   call physics_ptend_init(ptend_loc, state%psetcols, 'zm_convr', ls=.true., lq=lq, lu=doslop_uwind, lv=doslop_vwind)
+
+   !----------------------------------------------------------------------------
+   ! Associate pointers with physics buffer fields
 
-!
-! Associate pointers with physics buffer fields
-!
    itim_old = pbuf_old_tim_idx()
-   call pbuf_get_field(pbuf, cld_idx,         cld,    start=(/1,1,itim_old/), kount=(/pcols,pver,1/) )
-
-   call pbuf_get_field(pbuf, icwmrdp_idx,     ql )
-   call pbuf_get_field(pbuf, rprddp_idx,      rprd )
-   call pbuf_get_field(pbuf, fracis_idx,      fracis, start=(/1,1,1/),    kount=(/pcols, pver, pcnst/) )
-   call pbuf_get_field(pbuf, nevapr_dpcu_idx, evapcdp )
-   call pbuf_get_field(pbuf, prec_dp_idx,     prec )
-   call pbuf_get_field(pbuf, snow_dp_idx,     snow )
-
-
-! DCAPE-ULL
-   if(trigdcape_ull .or. trig_dcape_only)then
-     call pbuf_get_field(pbuf, t_star_idx,     t_star)
-     call pbuf_get_field(pbuf, q_star_idx,     q_star)
-     if ( is_first_step()) then
-       q_star(:ncol,:pver) = state%q(:ncol,:pver,1)
-       t_star(:ncol,:pver) = state%t(:ncol,:pver)
-     end if
-   end if
+   call pbuf_get_field(pbuf, cld_idx,           cld,    start=(/1,1,itim_old/), kount=(/pcols,pver,1/) )
+   call pbuf_get_field(pbuf, fracis_idx,        fracis, start=(/1,1,1/),        kount=(/pcols, pver, pcnst/) )
+   call pbuf_get_field(pbuf, icwmrdp_idx,       ql)
+   call pbuf_get_field(pbuf, rprddp_idx,        rprd)
+   call pbuf_get_field(pbuf, nevapr_dpcu_idx,   evapcdp)
+   call pbuf_get_field(pbuf, prec_dp_idx,       prec)
+   call pbuf_get_field(pbuf, snow_dp_idx,       snow)
+   call pbuf_get_field(pbuf, icimrdp_idx,       qi)
+   call pbuf_get_field(pbuf, dlfzm_idx,         dlf)
+   call pbuf_get_field(pbuf, difzm_idx,         dif)
 
-   call pbuf_get_field(pbuf, icimrdp_idx,     qi )
-   call pbuf_get_field(pbuf, dlfzm_idx,  dlf)
-   call pbuf_get_field(pbuf, difzm_idx,  dif)
+   ! DCAPE-ULL
+   if (trigdcape_ull .or. trig_dcape_only) then
+      call pbuf_get_field(pbuf, t_star_idx, t_star)
+      call pbuf_get_field(pbuf, q_star_idx, q_star)
+      if ( is_first_step()) then
+         q_star(1:ncol,1:pver) = state%q(1:ncol,1:pver,1)
+         t_star(1:ncol,1:pver) = state%t(1:ncol,1:pver)
+      end if
+   end if
 
    if (zm_microp) then
       call pbuf_get_field(pbuf, dnlfzm_idx, dnlf)
       call pbuf_get_field(pbuf, dnifzm_idx, dnif)
       call pbuf_get_field(pbuf, dsfzm_idx,  dsf)
       call pbuf_get_field(pbuf, dnsfzm_idx, dnsf)
-      call pbuf_get_field(pbuf, wuc_idx, wuc  )
+      call pbuf_get_field(pbuf, wuc_idx,    wuc)
    else
-      allocate(dnlf(pcols,pver), dnif(pcols,pver),dsf(pcols,pver), dnsf(pcols,pver))
-      allocate(wuc(pcols,pver))
+      allocate(dnlf(pcols,pver), &
+               dnif(pcols,pver), &
+               dsf(pcols,pver),  &
+               dnsf(pcols,pver), &
+               wuc(pcols,pver)   )
    end if
-   wuc(:,:) = 0._r8
+   wuc(:,:) = 0
```

Comment:
For consistency, it may be worth noting the array limits as has been done elsewhere.

Reply:
good catch, I'll add those.

Reply:
fixed.

Reply:
Thanks! Credit goes to Xiaoliang Song, who caught it.